src/SimpleRuleHandlerContainer.php

```diff
@@ -4,20 +4,28 @@
 
 namespace Yiisoft\Validator;
 
+use Yiisoft\Translator\TranslatorInterface;
 use Yiisoft\Validator\Exception\RuleHandlerInterfaceNotImplementedException;
 use Yiisoft\Validator\Exception\RuleHandlerNotFoundException;
 
 final class SimpleRuleHandlerContainer implements RuleHandlerResolverInterface
 {
     private array $instances = [];
 
+    private TranslatorInterface $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
     public function resolve(string $className): RuleHandlerInterface
     {
         if (!class_exists($className)) {
             throw new RuleHandlerNotFoundException($className);
         }
         if (!array_key_exists($className, $this->instances)) {
-            $classInstance = new $className();
+            $classInstance = new $className($this->translator);
```

Comment:
That's an assumption that a first arguemnt of a rule should be translator. Could it be improved?

Reply:
Made translator required, added using of named argument. Please take a look.